persistence.js

```diff
@@ -223,13 +226,18 @@ RedisPersistence.prototype.subscriptionsByClient = function (client, cb) {
 
   var clientSubKey = 'client:sub:' + client.id
 
-  pipeline.hgetall(clientSubKey, function (err, subs) {
-    var toReturn = Object.keys(subs).map(function (sub) {
-      return {
-        topic: sub,
-        qos: parseInt(this[sub])
-      }
-    }, subs)
+  pipeline.hgetall(clientSubKey, function returnSubs (err, subs) {
+    var subKeys = Object.keys(subs)
+
+    var toReturn = []
+
+    for (var i = 0; i < subKeys.length; i++) {
+      toReturn.push({
+        topic: subKeys[i],
+        qos: parseInt(subs[subKeys[i]])
+      })
+    }
```

Comment:
this should be moved to an external function, loops needs to be optimized separately.